### PR TITLE
Update to snapshot v3client

### PR DESCRIPTION
**Pull request by xing-yang**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
This PR updates snapshot dependency to v3/client.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
```release-note
Update to use snapshot v3/client
```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -7,7 +7,7 @@ require (
 	github.com/golang/mock v1.4.3
 	github.com/kubernetes-csi/csi-lib-utils v0.8.1
 	github.com/kubernetes-csi/csi-test/v3 v3.1.1
-	github.com/kubernetes-csi/external-snapshotter/client/v2 v2.2.0-rc3
+	github.com/kubernetes-csi/external-snapshotter/client/v3 v3.0.1-0.20201116082704-1e5a7ef7a73d
```
Comment (msau42) [inline go.mod:10]: do you know why there's a hash afterwards instead of just "v3.0.1"?

Comment (xing-yang) [reply go.mod:10]: Yes.  I do a go get with the snapshot commit `go get github.com/kubernetes-csi/external-snapshotter/client/v3@1e5a7ef7a73d691da0e6d350ba9613b05244d78f` and that will generate this hash automatically when it picks that commit.

Comment (xing-yang) [reply go.mod:10]: We didn't really cut a 3.0.1 release for the client so it will not be just 3.0.1.  The latest official release is 3.0.0 for the client.

Comment (msau42) [reply go.mod:10]: Can we import v3.0.0 instead?

Comment (xing-yang) [reply go.mod:10]: That's probably fine too.

**Pull request merged**